graph/number_of_islands_dfs.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file
+ *
+ * @brief Algorithm to count the number of islands
+ *
+ * @details
+ * Depth First Traversal (or Search) for a graph is similar to Depth First
+ * Traversal of a tree. The only catch here is, that, unlike trees, graphs may
+ * contain cycles (a node may be visited twice). To avoid processing a node more
+ * than once, use a boolean visited array. A graph can have more than one DFS
+ * traversal.
+ *
+ * The algorithm implemented in this file determines the number of islands in a
+ * 2D array of 1s(land) and 0s(water).
+ *
+ * @author [Harsh Agrawal](https://github.com/harshagrawal523)
+ *
+ */
+#include <iostream>
+#include <string>
+#include <vector>
+
+/**
+ * @namespace islands
+ * @brief DFS algorithms
+ */
+namespace islands {
+/**
+ * @namespace number_of_islands
+ * @brief Functions for counting the number of islands
+ */
+namespace number_of_islands {
+/**
+ * @brief Class for representing the matrix of 0s(water) and 1s(land).
+ */
+class matrix {
+ private:
+    int n, m;  // dimensions of graph
+    std::vector<std::vector<char> >
+        mat;  // matrix that stores the ocean(0s and 1s)
+
+ public:
+    /**
+     * @brief Constructor that initializes the matrix of oceans on creation
+     */
+    explicit matrix(std::vector<std::vector<char> > mat, int rows, int cols) {
+        n = rows;
+        m = cols;
+        mat.resize(n, std::vector<char>(m));
+    }
+    void dfs(std::vector<std::vector<char> > &mat, int row,
+             int col);  // dfs function that helps us find the number of islands
+};
+/**
+ * @brief this is a dfs funtion which checks all the connected 1s
+ * the function returns if it goes out of bound of matrix
+ */
+void matrix::dfs(std::vector<std::vector<char> > &matrix, int row, int col) {
```

Comment:
Here seems to be the issue. Check the CI logs https://github.com/TheAlgorithms/C-Plus-Plus/actions/runs/3169823215/jobs/5162010561#step:10:47

Reply:
I guess I have resolved this issue but something is still wrong and I have no idea what is wrong

Reply:
Can you check what's the issue? @Panquesito7 @mishraabhinn